crates/iota-data-ingestion/src/workers/blob.rs

```diff
@@ -2,31 +2,132 @@
         Self {
-            remote_store: create_remote_store_client(config.url, config.remote_store_options, 10)
-                .expect("failed to create remote store client"),
+            checkpoint_chunk_size_mb: config.checkpoint_chunk_size_mb,
+            remote_store: create_remote_store_client_with_ops(
+                config.url,
+                config.remote_store_options,
+                config.request_timeout_secs,
+                RetryConfig {
+                    max_retries: 10,
+                    retry_timeout: Duration::from_secs(config.request_timeout_secs + 1),
+                    backoff: BackoffConfig {
+                        init_backoff: Duration::from_secs(1),
+                        ..Default::default()
+                    },
+                },
+            )
+            .expect("failed to create remote store client"),
+        }
+    }
+
+    /// Uploads a Checkpoint blob to the Remote Store.
+    ///
+    /// If the blob size exceeds the configured `CHUNK_SIZE`,
+    /// it uploads the blob in parts using multipart upload.
+    /// Otherwise, it uploads the blob directly.
+    async fn upload_blob(&self, bytes: Vec<u8>, chk_seq_num: u64, location: Path) -> Result<()> {
+        if bytes.len() > self.checkpoint_chunk_size_mb as usize {
+            return self
+                .upload_blob_multipart(bytes, chk_seq_num, location)
+                .await;
         }
+
+        self.remote_store
+            .put(&location, Bytes::from(bytes).into())
+            .await?;
+
+        Ok(())
+    }
+
+    /// Uploads a large Checkpoint blob to the Remote Store using multipart
+    /// upload.
+    ///
+    /// This function divides the input `bytes` into chunks of size `CHUNK_SIZE`
+    /// and uploads each chunk individually.
+    /// Finally, it completes the multipart upload by assembling all the
+    /// uploaded parts.
+    async fn upload_blob_multipart(
+        &self,
+        bytes: Vec<u8>,
+        chk_seq_num: u64,
+        location: Path,
+    ) -> Result<()> {
+        let mut multipart = self.remote_store.put_multipart(&location).await?;
+        let chunks = bytes.chunks(self.checkpoint_chunk_size_mb as usize);
+        let total_chunks = chunks.len();
+
+        let mut parts_ordered_futures = chunks
+            .into_iter()
+            .map(|chunk| multipart.put_part(Bytes::copy_from_slice(chunk).into()))
+            .collect::<FuturesUnordered<_>>();
+
+        let mut uploaded_part = 0;
+        while let Some(part_result) = parts_ordered_futures.next().await {
+            match part_result {
+                Ok(()) => {
+                    uploaded_part += 1;
+                    tracing::info!(
+                        "uploaded checkpoint {chk_seq_num} chunk {uploaded_part}/{total_chunks}"
+                    );
+                }
+                Err(err) => {
+                    multipart.abort().await?;
+                    tracing::error!("error uploading part: {err}");
```

Comment:
Shouldn't we return the error here?

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5026/commits/a5f7d89854d05adaf7e7532fe729f1b748507e50#diff-2d923450ac72bc499c7474995e7f7811e449b1baa460bd5644c03b58f7d3d60dR127)